ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -11,51 +11,78 @@
 
 /*
  * ListBuckets --
- *     Builds a list of buckets from AWS account into a vector. Returns true if success, otherwise
- *     false.
+ *     Builds a list of buckets from AWS account into a vector. Returns 0 if success, otherwise
+ *     1.
  */
-bool
+int
 S3Connection::ListBuckets(std::vector<std::string> &buckets) const
```

Comment:
This can be a static class method.

Reply:
Agreed. Also, this method will go away in another few days, so be mindful of that.

Reply:
Ah no worries then!